spec/functional/resource/dnf_package_spec.rb

```diff
@@ -128,13 +127,114 @@ def pkg_arch
+        end
+
+        it "installs the rpm when #{vstring} is in the version property" do
+          flush_cache
+          dnf_package.package_name("chef_rpm")
+          dnf_package.version(vstring)
+          dnf_package.run_action(:install)
+          expect(dnf_package.updated_by_last_action?).to be true
+        end
+
+        it "is idempotent when #{vstring} is in the version property" do
+          preinstall("chef_rpm-1.10-1.#{pkg_arch}.rpm")
+          dnf_package.package_name("chef_rpm")
+          dnf_package.version(vstring)
+          dnf_package.run_action(:install)
+          expect(dnf_package.updated_by_last_action?).to be false
+        end
+
+        it "upgrades the rpm when #{vstring} is in the package_name" do
+          flush_cache
+          dnf_package.package_name("chef_rpm-#{vstring}")
+          dnf_package.run_action(:upgrade)
+          expect(dnf_package.updated_by_last_action?).to be true
+        end
+
+        it "is idempotent when #{vstring} is in the package_name" do
+          preinstall("chef_rpm-1.10-1.#{pkg_arch}.rpm")
+          dnf_package.package_name("chef_rpm-#{vstring}")
+          dnf_package.run_action(:upgrade)
+          expect(dnf_package.updated_by_last_action?).to be false
+        end
+
+        it "upgrades the rpm when #{vstring} is in the version property" do
+          flush_cache
+          dnf_package.package_name("chef_rpm")
+          dnf_package.version(vstring)
+          dnf_package.run_action(:upgrade)
+          expect(dnf_package.updated_by_last_action?).to be true
+        end
+
+        it "is idempotent when #{vstring} is in the version property" do
+          preinstall("chef_rpm-1.10-1.#{pkg_arch}.rpm")
+          dnf_package.package_name("chef_rpm")
+          dnf_package.version(vstring)
+          dnf_package.run_action(:upgrade)
+          expect(dnf_package.updated_by_last_action?).to be false
+        end
+      end
+
+      %w{1.2 1* 1.2-1 1*-1 1.2-* 1*-* 0:1.2 0:1* 0:1.2-1 0:1*-1 *:1.2-* *:1*-*}.each do |vstring|
+        it "is idempotent when #{vstring} is in the version property and there is a candidate version" do
+          preinstall("chef_rpm-1.2-1.#{pkg_arch}.rpm")
+          dnf_package.package_name("chef_rpm")
+          dnf_package.version(vstring)
+          dnf_package.run_action(:install)
+          expect(dnf_package.updated_by_last_action?).to be false
+          expect(shell_out("rpm -q --queryformat '%{NAME}-%{VERSION}-%{RELEASE}.%{ARCH}\n' chef_rpm").stdout.chomp).to match("^chef_rpm-1.2-1.#{pkg_arch}$")
+        end
+      end
+
+      %w{1.2 1.2-1 1.2-* 0:1.2 0:1.2-1 *:1.2-*}.each do |vstring|
+        it "is idempotent when #{vstring} is in the version property on upgrade and it doesn't match the candidate version" do
+          preinstall("chef_rpm-1.2-1.#{pkg_arch}.rpm")
+          dnf_package.package_name("chef_rpm")
+          dnf_package.version(vstring)
+          dnf_package.run_action(:upgrade)
```

Comment:
this is a pretty big change... you're now saying that:
```
package 'foo' do
  version '1.2'
  action :install # or :upgrade
end
```

will work and install the latest-release of 1.2. That needs to be called out BIG TIME in the release notes. Before that would break.

Reply:
that's the bug that is being fixed.

Reply:
I mean... sorta? You had to be explicit before. The bug you're fixing is that specifying arch in weird places would lead to lack of idempotency.
I'm not saying this is wrong, but this is definitely gonna lead to some unexpected surprises for people.